server/metrics/metrics.go

```diff
@@ -286,6 +297,11 @@ func (m *metrics) ObserveLinkedChannelsTotal(count int64) {
 		m.linkedChannelsTotal.Set(float64(count))
 	}
 }
+func (m *metrics) ObserveUpstreamUsersTotal(count int64) {
+	if m != nil {
```

Comment:
why do we need this check here and not when we are actually calling this ObserveUpstreamUsersTotal

Reply:
this is fixed in PR https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/369 where we have added checks where the function is called